Development/nmos/sdp_utils.h

```diff
@@ -82,19 +83,35 @@ namespace nmos
         struct origin_t
         {
             utility::string_t user_name;
-            uint64_t session_id;
-            uint64_t session_version;
+            utility::string_t session_id;
+            utility::string_t session_version;
 
             origin_t() : session_id(), session_version() {}
             origin_t(const utility::string_t& user_name, uint64_t session_id, uint64_t session_version)
                 : user_name(user_name)
-                , session_id(session_id)
-                , session_version(session_version)
+                , session_id(boost::lexical_cast<utility::string_t>(session_id))
+                , session_version(boost::lexical_cast<utility::string_t>(session_version))
             {}
-            origin_t(const utility::string_t& user_name, uint64_t session_id_version)
+            origin_t(const utility::string_t& user_name, const utility::string_t& session_id, const utility::string_t& session_version)
                 : user_name(user_name)
-                , session_id(session_id_version)
-                , session_version(session_id_version)
+                , session_version(session_version)
```

Comment:
You've still got a member initializer here as well as the code in the constructor body...

Might be nice to use the member initializers for both, like this:

```
const utility::string_t& valid_numeric_string(const utility::string_t& s)
{
  if (!std::all_of(s.begin(), s.end(), ::isdigit))
  {
    throw std::invalid_argument("not a numeric string");
  }
  return s;
}
...
  , session_id(valid_numeric_string(session_id))
  , session_version(valid_numeric_string(session_version))
```
Although you do lose the precision of the error message of course.